### DISCUSSION TITLE
Restrict user permissions to see/edit user groups

### DISCUSSION THREAD

**Discussion by joelzerner** [craftcms/cms#8682] [Ideas]
### Description

Under the "Users" permissions group, it would be great to add permissions to restrict a user to only be able to see/edit users from selected groups. Currently, we are able to give a user permissions to only assign users to certain groups, however they are still able to (a) see all users, (b) edit all users and (c) remove a user from a user group they do have permissions to assign for. 

### Steps to reproduce

1. Create two user groups - "Master User" and "Servant User"
2. Create two users, one for each group
3. Assign the Servant User permissions to be able to: edit users, register users, moderate users, assign user groups, assign users to servant user group.
4. Login as the Servant User
5. View the users list (in reference to issues (a) and (b) described above)
6. Edit the Master User (in reference to issue (c) described above)

### Similar issues?

[https://github.com/craftcms/cms/issues/1188](url)
[https://github.com/craftcms/cms/issues/3411](url)

### Additional info

- Craft version: 3.1.26

**Comment by acalvino4** [comment]
Absolutely! There was a plugin "Hide Admin" that at least restricted other users' ability to see/edit super admins, but it would be really nice to have this available in the more fine-grained manner that @joelzerner suggests.
